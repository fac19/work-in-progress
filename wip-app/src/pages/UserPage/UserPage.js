import React from "react";
import { getUser, getUserPageProjects } from "../../utils/get-fetch";
import { ProfileCard } from "./UserPage.style";
import ProjectFeed from "../../components/ProjectFeed/ProjectFeed";

const UserPage = (props) => {
  const [userInfo, setUserInfo] = React.useState([]);
  const [userProjects, setUserProjects] = React.useState([]);

  React.useEffect(() => {
    getUser().then((userInfo) => {
      setUserInfo(userInfo);
    });
  }, []);

  React.useEffect(() => {
    getUserPageProjects().then((userProjects) => {
      setUserProjects(userProjects.reverse());
    });
  }, []);

  const {
    username,
    email,
    user_bio,
    user_vocation,
    user_link_1,
    user_link_2,
    user_link_3,
  } = userInfo;

  const isLoading = userInfo.length === 0 || userProjects.length === 0;

  if (isLoading) {
    return <h1>Loading your profile...</h1>;
  } else {
    return (
      <>
        <ProfileCard>
          <h1>{username}'s Profile</h1>
          <p>{email}</p>
          <a
            href={"https://" + user_link_1}
            target="_blank"
            rel="noopener noreferrer"
          >
            Link 1
          </a>
          <a
            href={"https://" + user_link_2}
            target="_blank"
            rel="noopener noreferrer"
          >
            Link 2
          </a>
          <a
            href={"https://" + user_link_3}
            target="_blank"
            rel="noopener noreferrer"
          >
            Link 3
          </a>
          <p>{user_bio}</p>
          <p>{user_vocation}</p>
        </ProfileCard>
        <ProjectFeed projects={userProjects} />
      </>
    );
  }
};

export default UserPage;
