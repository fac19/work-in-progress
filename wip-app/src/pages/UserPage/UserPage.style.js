import styled from "styled-components";

const ProfileCard = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-items: center;
`;

const ProfileName = styled.h1`
  margin-bottom: 1em;
  font-size: 2em;
  color: rgba(0, 0, 0, 0.7);
`;

const ProfileEmail = styled.p`
  font-size: 1em;
  color: rgba(0, 0, 0, 0.6);
`;
const ProfileBio = styled.p`
  font-size: 1em;
  color: rgba(0, 0, 0, 0.6);
`;
const ProfileVocation = styled.p`
  font-size: 1em;
  color: rgba(0, 0, 0, 0.6);
`;
const ProfileATag = styled.a`
  margin: 0;
`;
const ProfileLink = styled.img`
  width: 2em;
`;
const ProfileLinksSection = styled.section`
  display: flex;
  flex-direction: rows;
  width: 2em;
  justify-content: center;
`;

export {
  ProfileCard,
  ProfileName,
  ProfileEmail,
  ProfileBio,
  ProfileVocation,
  ProfileATag,
  ProfileLink,
  ProfileLinksSection,
};
