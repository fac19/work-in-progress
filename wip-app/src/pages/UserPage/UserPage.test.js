import React from "react";
import TestRenderer from "react-test-renderer";
import UserPage from "./UserPage";

describe("Snapshot test", () => {
  it("renders the correct ui", () => {
    const component = TestRenderer.create(<UserPage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
