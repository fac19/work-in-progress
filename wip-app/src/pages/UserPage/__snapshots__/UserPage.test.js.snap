// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<h1
  className="sc-fznKkj fbGqhE"
>
  Loading your profile...
</h1>
`;
